app/presenters/reservation_calendar.rb

```diff
@@ -0,0 +1,31 @@
+class ReservationCalendar < SimpleDelegator
+
+  attr_accessor :hostname
+
+  def self.to_calendar(reservation)
+    ReservationCalendar.new(reservation).as_ical
+  end
+
+  def initialize(reservation, hostname)
+    super(reservation)
+    @hostname = hostname
```

Comment:
You can get the host name from the request, but this presenter won't have the request, I assumed that the controller that was creating this object would pass the hostname in.